components/script/script_runtime.rs

```diff
@@ -931,27 +931,19 @@ unsafe fn set_gc_zeal_options(cx: *mut RawJSContext) {
 #[cfg(not(feature = "debugmozjs"))]
 unsafe fn set_gc_zeal_options(_: *mut RawJSContext) {}
 
-#[derive(Clone, Copy)]
-#[repr(transparent)]
-pub(crate) struct JSContext(*mut RawJSContext);
+pub(crate) use script_bindings::script_runtime::JSContext;
 
-#[allow(unsafe_code)]
-impl JSContext {
-    /// Create a new [`JSContext`] object from the given raw pointer.
-    ///
-    /// # Safety
-    ///
-    /// The `RawJSContext` argument must point to a valid `RawJSContext` in memory.
-    pub(crate) unsafe fn from_ptr(raw_js_context: *mut RawJSContext) -> Self {
-        JSContext(raw_js_context)
-    }
+pub(crate) trait JSContextHelper {
```

Comment:
Perhaps document this as being necessary because of `SEEN_POINTERS`?

Reply:
I just thought it was necessary because the type doesn't live in the same crate anymore.